Add support for more JSON Schema specs

dfernandez-at-wiris opened the pull request:
- add checks for drafts 4, 6 and 7, defaulting to 7 if unspecified
- simplify exceptions, reduce them in number
- add a minor functionality to list the available configurations
- minor docstring and README.md corrections/expansions
- add the appropriate unit tests & fixtures for the added functionality

dfernandez-at-wiris committed:
Add support for more JSON Schema specs
- add checks for drafts 4, 6 and 7, defaulting to 7 if unspecified
- simplify exceptions, reduce them in number
- minor docstring and README.md corrections/expansions
- add the appropriate unit tests & fixtures for the added functionality

hqiu-at-wiris commented on santoku/utils/README.md:
```diff
@@ -7,44 +7,38 @@ The `configuration` module provides classes to store and manage configuration fi
 
 ### Schema validation
-For schema validation of configuration files, we use our own definition of what a configuration looks like and also our own definition of what a schema should look like. In short, our schema is a narrow subset of [JSON Schema](https://json-schema.org/). We plan on fully adopting JSON Schema for our validation needs, but due to the lack of native Python tools and the effort of implementing our own ...
+For schema validation of configuration files, we use [JSON Schema](https://json-schema.org/) and the [jsonschema](https://pypi.org/project/jsonschema/) package. For a 
```
Is this "For a" a typo?

dfernandez-at-wiris replied:
Indeed

hqiu-at-wiris commented on santoku/utils/configuration.py:
```diff
@@ -137,13 +122,15 @@ def __init__(
         initial_configuration: str, optional
             Name of the configuration in the `configurations` that will be set as the initial one.
 
-        InvalidConfiguration
```
Missing `Raises` section title.

hqiu-at-wiris commented on santoku/utils/configuration.py:
```diff
@@ -2,7 +2,6 @@
 
-from jsonschema import Draft7Validator
 from jsonschema.exceptions import ValidationError, SchemaError
 from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
```
`Iterable` and `Tuple` not needed.

hqiu-at-wiris commented on santoku/utils/test/test_configuration.py:
```diff
@@ -5,13 +5,13 @@
 from collections import namedtuple
```
Not used anywhere.

dfernandez-at-wiris committed:
Minor fixes to utils.configuration
- fix outdated markdown docs at the subpackage level
- fix some typos in docstrings
- remove some unused imports
